### PR TITLE
Add active CARLOS menu tab indicator

### PR DESCRIPTION
**Pull request by CecilFoubert**
## Summary
- mark the current CARLOS menu tab with a nav-active class based on the current request path
- style the active tab with the same blue background used by nav hover states
- keep alert/count text white while its tab is active so the active state is readable
- add regression coverage for active-tab markup and CSS

refs #2254

## Tests
- mvn -q -Dtest=ScheduleNavigationAssetRegressionTest test

---
## Summary by cubic
Highlights the active CARLOS top-nav tab based on the current or forwarded path. Keeps Inbox highlighted when opened from Schedule and aligns styling/layout across the top nav, schedule header, and Inbox.

- New Features
  - Server-side active-tab detection for Schedule, Search, Inbox, Tickler, Messenger, Consultation, Documents, Reports, Admin, and eConsult; suppress Documents when Inbox is active.
  - Apply nav-active in `mainMenu.jsp` and the schedule header; route Inbox/Unclaimed to real URLs with `scheduleNav=1`; include the shared top nav and load `topnav.css` in Inbox when opened from Schedule.

- Refactors
  - Extract path matching to `io.github.carlos_emr.carlos.utility.NavPath`; update JSPs to `NavPath.requestPathMatches()` and add `NavPathUnitTest`, including coverage for forward request URI.
  - Normalize top-nav layout/styles: move logo/logout into the first menu cell; scope spacing/hover to direct children (`ul#navlist > li` and `> a`); unify active/hover/focus and `.nav-active` styles in `topnav.css` and receptionist schedule CSS; remove the legacy Resources menu.
  - Use `openScheduleSection`/`openScheduleMenuSection` for Inbox/menu actions with popup fallbacks; expand regression tests to lock in selector structure, nav-active behavior, Inbox/schedule integration, path-boundary cases, and forward-path handling.

<sup>Written for commit 421ec9a03b8ea2d7d6b25d579f2d66b0fc174e4d. Summary will update on new commits.</sup>

<a href="https://cubic.dev/pr/carlos-emr/carlos/pull/2243?utm_source=github" target="_blank" rel="noopener noreferrer" data-no-image-dialog="true"><picture><source media="(prefers-color-scheme: dark)" srcset="https://cubic.dev/buttons/review-in-cubic-dark.svg"><source media="(prefers-color-scheme: light)" srcset="https://cubic.dev/buttons/review-in-cubic-light.svg">Review in cubic</picture></a>

## Summary by Sourcery

Highlight the active CARLOS top navigation tab based on the current or forwarded request path and align schedule, inbox, and receptionist views with consistent layout and styles.

New Features:
- Add server-side detection of active nav tabs for schedule, search, inbox, tickler, messenger, consultation, documents, reports, admin, and eConsult based on request and forward paths.
- Render the shared top navigation bar and styling within Inboxhub when opened in schedule navigation mode so the Inbox respects active-tab state.

Enhancements:
- Unify nav item layout and active/hover styling for the top navigation and receptionist schedule header, including updated logo/logout placement and scoped list item rules.
- Route Inbox and unclaimed lab links through real URLs in schedule views so active state and scheduleNav integration are preserved while still supporting popup behavior.
- Remove the legacy clinical resources menu entries from provider navigation to simplify the top nav.

Tests:
- Extend schedule navigation regression tests to cover active-tab markup and CSS contracts, Inboxhub scheduleNav integration, NavPath usage in JSPs, and receptionist/topnav styling.
- Introduce NavPath unit tests to validate boundary-aware path matching and handling of Jakarta forward attributes.

### CODE REVIEW

**Comment by CecilFoubert** [comment]
image
image
image
Screenshots for clarity, managed to get the active highlight in, also took out the Resources tab.

**Comment by yingbull** [comment]
@CecilFoubert  not sure if ready for review is an accident, but I am setting it back to draft due to the review comments open.

**Comment by CecilFoubert** [comment]
Agh, those weren't here yet when I put it for review, I'll fix those and see if it has anything else to poke me about.

**Comment by CecilFoubert** [comment]
The open sourcery-ai review thread about the dashboard dropdown selector leak (comment on `topnav.css` line 63) is addressed — the current code at `topnav.css:61` and `receptionistapptstyle.css:443` both use `ul#navlist > li:not(.dashboardDropdown) > a` (child combinator, class on `li`) which is the correct form. The selector `a:not(.dashboardDropdown)` from the earlier commit that triggered the comment was replaced in commit 4dd6875. The thread can be resolved.

Separately, the `Fixes #2240` in the PR description has been updated to `refs #2254` since #2240 was superseded by umbrella #2254. Linking to #2240 was safe (it is already closed so nothing auto-closes), but #2254 is the active tracking issue.

The remaining failing check is DCO — every commit on this branch needs a `Signed-off-by: Cecil Foubert <email>` trailer. Because this requires rebasing the branch history, it cannot be done automatically here — please run:
```
git rebase --signoff origin/develop
git push
```
against the branch to add sign-offs to all commits at once.

**Comment by CecilFoubert** [comment]
Ok, long story short, ready for review, but once it's in I can play with my other pr (2242, the menu consistency one) and get them synced up properly. Ready to be reviewed @yingbull.

**Comment by CecilFoubert** [comment]
Alright, all the little things caught in review are handled, should be good to merge to main @yingbull

**Comment by yingbull** [comment]
@CecilFoubert You can use Codex to help signoff with -s on the branch and push it back to clear DCO; you might need to click on the DCO Sing-off ... button to tell it to run the failed test again, but doing it that way should clear.   There are also unresolved review comments on the PR.  Please address and sednback to review.

File: src/main/webapp/WEB-INF/jsp/provider/mainMenu.jsp [hunk 1]
```diff
@@ -84,6 +84,27 @@
     String userlastname = loggedInProvider != null ? loggedInProvider.getLastName() : "";
     String encodedUserName = URLEncoder.encode(StringUtils.trim(userfirstname + " " + userlastname), StandardCharsets.UTF_8);
     boolean scheduleNavActive = "1".equals(request.getParameter("scheduleNav"));
+    String navRequestPath = StringUtils.defaultString(request.getRequestURI()) + " "
+            + StringUtils.defaultString(request.getServletPath()) + " "
+            + StringUtils.defaultString((String) request.getAttribute("jakarta.servlet.forward.request_uri")) + " "
+            + StringUtils.defaultString((String) request.getAttribute("jakarta.servlet.forward.servlet_path"));
+    boolean scheduleTabActive = navRequestPath.contains("/provider/providercontrol")
+            || navRequestPath.contains("appointmentprovideradmin");
```
[the remark this answers is not in the document]
Comment (CecilFoubert) [reply src/main/webapp/WEB-INF/jsp/provider/mainMenu.jsp]: Addressed by the `NavPath` utility class. `forwardAttribute()` uses pattern-matching `instanceof` (no ClassCastException possible), `pathMatches()` enforces a slash/query-char boundary (no false partial-segment matches), all patterns carry leading slashes, and the varargs loop replaces the concatenated string.

File: src/main/webapp/WEB-INF/jsp/provider/mainMenu.jsp [hunk 2]
```diff
@@ -84,6 +84,27 @@
...
+    boolean messengerTabActive = navRequestPath.contains("/messenger/");
+    boolean consultationTabActive = navRequestPath.contains("/encounter/IncomingConsultation")
+            || navRequestPath.contains("/encounter/oscarConsultationRequest");
+    boolean documentTabActive = navRequestPath.contains("/documentManager/") && !inboxTabActive;
+    boolean reportTabActive = navRequestPath.contains("/report/")
+            || navRequestPath.contains("/oscarReport/");
+    boolean adminTabActive = navRequestPath.contains("/administration")
+            || navRequestPath.contains("/admin/");
+    boolean resourceTabActive = navRequestPath.contains("/resource");
+    boolean econsultTabActive = navRequestPath.contains("/encounter/econsult");
```
[the remark this answers is not in the document]
Comment (CecilFoubert) [reply src/main/webapp/WEB-INF/jsp/provider/mainMenu.jsp]: Addressed — all patterns now pass through `NavPath.pathMatches()` which requires a segment boundary after the match (next char must be `/`, `?`, `;`, or `#`). Patterns ending with `/` are accepted as-is. `/administration` is now checked against boundary chars so `/administration-panel` does not match. `/econsult` is scoped to `/encounter/econsult` so unrelated paths are not triggered.

File: src/main/webapp/WEB-INF/jsp/provider/appointmentprovideradminday.jsp [hunk 3]
```diff
@@ -903,14 +903,14 @@
                 <ul id="navlist">
                     <c:if test="${infirmaryView_isOscar != 'false'}">
                         <% if (request.getParameter("viewall") != null && request.getParameter("viewall").equals("1")) { %>
-                        <li>
+                        <li class="nav-active">
```
[the remark this answers is not in the document]
Comment (CecilFoubert) [reply src/main/webapp/WEB-INF/jsp/provider/appointmentprovideradminday.jsp]: Addressed — fixed in commit 1f5b0057. `nav-active` is now computed dynamically via `NavPath.requestPathMatches(request, "/provider/providercontrol", "/provider/appointmentprovideradmin")` and stored in `scheduleNavActiveClass`, keeping the schedule header consistent with mainMenu.jsp.

File: src/main/webapp/css/topnav.css [hunk 4]
```diff
@@ -51,18 +52,22 @@ table#firstTable td#userSettings ul#userSettingsMenu:not(.dashboardDropdown) {
...
+table#firstTable td#firstMenu ul#navlist li:not(.dashboardDropdown) {
     line-height: 1;
     display: inline-block;
+    padding: 0;
 }
 
-table#firstTable td#firstMenu ul#navlist > li > a {
+table#firstTable td#firstMenu ul#navlist li a:not(.dashboardDropdown) {
     display: inline-block;
     padding: 5px;
```
[the remark this answers is not in the document]
Comment (CecilFoubert) [reply src/main/webapp/css/topnav.css]: Addressed — current code uses `ul#navlist > li:not(.dashboardDropdown)` and `ul#navlist > li:not(.dashboardDropdown) > a` (child combinator, `:not` on the `li`). The `a:not(.dashboardDropdown)` form that matched anchors nested inside `.dashboardDropdown` items is no longer present.

**Comment by CecilFoubert** [comment]
All review threads addressed:

- **Gemini (mainMenu.jsp — ClassCastException / loose matching / missing slash):** Fixed by the `NavPath` utility — `forwardAttribute()` uses pattern-matching `instanceof`, `pathMatches()` enforces segment boundaries, all patterns have leading slashes.
- **Gemini (mainMenu.jsp — broad admin/econsult patterns):** `pathMatches()` boundary checking prevents `/administration-panel` from matching `/administration`; `econsult` is scoped to `/encounter/econsult`.
- **Gemini (appointmentprovideradminday.jsp — hardcoded nav-active):** Fixed in commit `1f5b0057` — now uses `NavPath.requestPathMatches()` consistent with mainMenu.jsp.
- **Sourcery (topnav.css — dropdown anchor selector leak):** Fixed — current selectors use `> li:not(.dashboardDropdown) > a` (child combinator, class on `li`).

DCO: several commits on this branch are missing `Signed-off-by`. Please post **"Confirming DCO sign off for all commits"** to clear the DCO check, or re-sign the commits locally with `git rebase --signoff` and force-push.

**Comment by CecilFoubert** [comment]
Final sign-off on the review cleanup for this PR.

All previously unresolved review threads have been resolved, the branch was cleaned up to a single signed-off commit (`8a8cb9a`), and the focused validation passed:

```text
mvn -q -Dtest=NavPathUnitTest,ScheduleNavigationAssetRegressionTest test
```

Signed-off-by: Cecil Foubert <Cfoubert0782@conestogac.on.ca>

**Comment by yingbull** [comment]
@copilot+claude-opus-4.8 lgtm? Any other findings not addressed or found in your own review?

**Review by yingbull — APPROVED** [review]
